app/helpers/reactjs_helper.rb

```diff
@@ -65,6 +65,17 @@ def css_tags_for(requested_plugins)
     end
   end
 
+  def locale_js_tags
+    locale = FastGettext.locale
+    domains = ::Foreman::Plugin.all.filter_map do |plugin|
+      domain = plugin.gettext_domain
+      domain if (FastGettext.translation_repositories[domain]&.available_locales || []).include? locale
+    end
+    domains.map do |domain|
+      javascript_include_tag("locale/#{locale}/#{domain}")
+    end.join.html_safe
```

Comment:
Untested, but isn't it also needed to check if domain is `nil`? Or do you count on `FastGettext.translation_repositories[nil` returning `nil`? I was thinking about something like this:

```suggestion
    ::Foreman::Plugin.all.filter_map do |plugin|
      domain = plugin.gettext_domain
      if domain && (FastGettext.translation_repositories[domain]&.available_locales || []).include? locale
        javascript_include_tag("locale/#{locale}/#{domain}")
    end
  end.join.html_safe
```

Reply:
> Or do you count on FastGettext.translation_repositories[nil returning nil

Yes, I did that. The assumption seems to hold, what you suggest is hands down more obvious and probably safe in the long run. It needed a small tweak (parentheses) though